src/app/charts/extra-params-components/basetemp.component.ts

```diff
@@ -0,0 +1,64 @@
+import { AfterViewInit, Component, EventEmitter, Input, Output, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+import { Indicator } from '../../models/indicator.model';
+import { BasetempIndicatorQueryParams } from '../../models/basetemp-indicator-query-params.model';
+
+import * as _ from 'lodash';
+
+/*
+ * Basetemp params component
+ * Multi-field form to allow user to specify the basetemp params
+ */
+@Component({
+  selector: 'ccl-basetemp-parameters',
+  templateUrl: './basetemp.component.html'
+})
+export class BasetempComponent implements AfterViewInit, OnInit {
+
+    @Input() indicator: Indicator;
+    @Input() extraParams: BasetempIndicatorQueryParams;
+    @Output() basetempParamSelected = new EventEmitter<BasetempIndicatorQueryParams>();
+
+    basetempForm: FormGroup;
+
+    // default form values
+    private defaultBasetemp = 50;
+    private defaultBasetempUnit = 'F';
+
+    public basetempUnits: any[] = [
```

Comment:
Would like to see these shared across temperature unit fields, as we have several now, but this is fine for now.

Reply:
Good point. I'll merge this, and do that in the percentile params PR after a rebase where I'm still working on stuff